queries/leo/highlights.scm

```diff
@@ -0,0 +1,262 @@
+(identifier) @variable 
+
+[
+ "assert"
+ "assert_eq"
+ "assert_neq"
+ "block"
+ "console"
+ "const"
+ "constant"
+ "finalize"
+ "in"
+ "let"
+ "mapping"
+ "private"
+ "program"
+ "public"
+ "record"
+ "self"
+ "struct"
+] @keyword
+
+[
+ "transition"
+ "function"
+ "inline"
+] @keyword.function
+
+"import" @include
+
+"return" @keyword.return
+
+"for" @repeat
+
+[ 
+  "if"
+  "then"
+  "else"
+] @conditional
+
+[
+ (ternary_if)
+ (ternary_else)
+] @conditional.ternary
+
+
+[ ";" "," "::"] @punctuation.delimiter
+
+[
+"!"
+
+"&&"
+"||"
+
+"=="
+"!="
+
+"<"
+"<="
+">"
+">="
+
+"&"
+"|"
+"^"
+
+"<<"
+">>"
+
+"+"
+"-"
+"*"
+"/"
+"%"
+"**"
+
+"="
+
+"+="
+"-="
+"*="
+"/="
+"%="
+"**="
+
+"<<="
+">>="
+
+"&="
+"|="
+"^="
+
+"&&="
+"||="
+
+] @operator
+
+(comment) @comment
+
+(boolean_literal) @boolean
+
+(constant_declaration 
+  (identifier) @constant
+)
+
+;program name if with right extension "aleo" -> okay
+(program_declaration
+  (program_id 
+    name: (identifier) 
+    extension: (identifier) @_ext (#eq? @_ext "aleo")) @string
+)
+
+;program name if with not with extension "aleo" -> error
+(program_declaration
+  (program_id 
+    name: (identifier) 
+    extension: (identifier) @_ext (#not-eq? @_ext "aleo")) @punctuation.delimiter
+)
```

Comment:
If any extension is syntactically valid it should be highlighted the same.
If it's not syntactically valid it should be handled by the parser itself.
```suggestion
(program_declaration
  (program_id) @string.special)
```